src/main/java/org/jenkinsci/plugins/mesos/JenkinsScheduler.java

```diff
@@ -753,6 +768,7 @@ private void getContainerInfoBuilder(Offer offer, Request request, String slaveN
                   .newBuilder()
                   .setName("ports")
                   .setType(Value.Type.RANGES)
+                  .setRole(roleToUse)
```

Comment:
I've amended the PR to use the default value for the mesos role, in case nothing is set by Mesos.